app/models/services/facebook.rb

```diff
@@ -23,16 +23,16 @@ def post_to_facebook(url, body)
   end
 
   def create_post_params(post)
-    message = post.message.plain_text_without_markdown
-    if post.photos.any?
-      message += " " + short_post_url(post, protocol: AppConfig.pod_uri.scheme,
-                                            host: AppConfig.pod_uri.authority)
+    post_message = post.message.plain_text_without_markdown
+
+    if post_message.empty? && post.photos.none?
+      message = post_message
+    else
+      post_url = short_post_url(post, protocol: AppConfig.pod_uri.scheme, host: AppConfig.pod_uri.authority)
+      message = "#{post_message} (via #{post_url} )"
```

Comment:
When there is no text but only photos, I think `post_message` is empty. In this case I think it looks weird to have the post be just a "(via https://pod.example.org/p/1234)", so I think it's better to just post the link in that case, as it was before. As soon as the message isn't empty, add it with (via ...).

Also, does facebook really need the space before the closing brace to detect the link? As you can see above, GitHub and also diaspora does detect URLs perfectly without this space.

Reply:
Yes, please. It's already the message that is going to be posted to facebook, you only add some more things to it before you post it.